### PR TITLE
Optional icon for ListItem

### PR DESCRIPTION
**Pull request by judastheo**
## Checklist before requesting a review
- [x] Working on iOS
- [x] Working on Android
- [x] Integration tests added
- [x] Visual regressions screenshots are up to date
- [x] Design validated

### CODE REVIEW

File: packages/sandbox/src/app/sandbox/docs/list-item.doc.tsx [hunk 1]
```diff
@@ -8,6 +8,12 @@ const ICON_NAME = 'Area';
 const LONG_DESCRIPTION =
   'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt.';
 const ON_PRESS = () => Alert.alert('item pressed');
+const ASYNC_ON_PRESS = async () =>
+  new Promise((resolve) =>
+    setTimeout(() => {
+      resolve(undefined);
+    }, 2000)
+  );
```
Comment (ludwig-pro) [inline packages/sandbox/src/app/sandbox/docs/list-item.doc.tsx]: We should create a folder with utilities...

I have my own sleep function too 🙈 in `streamline/packages/sandbox/src/app/sandbox/docs/card-summary.doc.tsx`

```
  const sleep = async () =>
    new Promise((resolve) =>
      setTimeout(() => {
        resolve(undefined);
      }, 2000)
    );
 ```

Comment (ludwig-pro) [reply packages/sandbox/src/app/sandbox/docs/list-item.doc.tsx]: same for on_press with an alert?

File: packages/sandbox/e2e/ios/screenshots/ListItem.png
Comment (ludwig-pro) [inline packages/sandbox/e2e/ios/screenshots/ListItem.png:1]: humm we should improve the visual regression to take a picture of all the variation 🤔

**Review by ludwig-pro — APPROVED** [review]

**Comment by ianrobertlambert** [comment]
@judastheo Just to confirm, does the divider span from the start of the text? If yes, then we're good to go. :) Thanks a lot!

**Pull request merged**